go/stats/prometheusbackend/prometheusbackend.go

```diff
@@ -67,8 +64,11 @@ func (be PromBackend) publishPrometheusMetric(name string, v expvar.Var) {
 		newMultiTimingsCollector(st, be.buildPromName(name))
 	case *stats.Histogram:
 		newHistogramCollector(st, be.buildPromName(name))
+	case *stats.String, stats.StringFunc, stats.StringMapFunc, *stats.Rates:
+		// silently ignore these types since they don't make sense to
+		// export to prometheus' data model
 	default:
-		logUnsupported.Infof("Not exporting to Prometheus an unsupported metric type of %T: %s", st, name)
+		log.Warningf("Not exporting to Prometheus an unsupported metric type of %T: %s", st, name)
```

Comment:
How about log.Fatalf here instead?

Some background:

We have similar internal code.

In the past, we did only log a warning similar to this.

However, I plan to change this and issue a log.Fatalf() here instead. That's because a crash in a place like this would have caught the missed type handling I accidentally introduced in our internal code (when I adjusted it to the types refactor).

Given that, you probably want to do a similar thing here as well?

A panic here bears the risk that some code doesn't register their variable during init() and instead dynamically at runtime. But as long as each type has at least one variable instance which publishes during/near init(), we will detect such crashes in tests and not in production. That should be good enough?

For our internal plugins, I'll also guard this behavior by a flag and turn the flag off in our production instance config only. You could also add a flag here, but I would rather not set such a flag in e.g. our Kubernetes config. It's just too confusing.

Given all that, I suggest to just change this to log.Fatalf.

Reply:
@michael-berlin I'm on board with that change and agree the future-proofing benefit is worth the risk.

I opted to _not_ add another flag here and as a result we need to be vigilant when other stats types are refactored.

Then again until we did this work, I don't think anyone touched stats in the past year at least, so I think this is pretty safe all things considered.